src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
         except Exception as e:
             raise SystemInfoFetchError(f"Unable to fetch System Info") from e
 
+@dataclass
+class TrainingDetails:
+    hyperparams: dict = None
+    loss_func: str = ""
+    optimizer: str = ""
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["hyperparams"] = self.hyperparams
+        model_card["loss_func"] = self.loss_func
+        model_card["optimizer"] = self.optimizer
+        return model_card
+
+@dataclass
+class EvalDetails:
+    loss_funcs: list = None
+    metrics: list = None
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["loss_funcs"] = self.loss_funcs
+        model_card["metrics"] = self.metrics
+        return model_card
+
+@dataclass
+class ModelCard:
+    model_details: ModelDetails
+    dataset_details: DatasetDetails
+    experiment_environment: ExperimentEnvironment
+    training_details: TrainingDetails
+    eval_details: EvalDetails
+
+    def toJSON(self) -> Dict:
+        model_card = {}
+        model_card["model_details"] = self.model_details.toJSON()
+        model_card["dataset_details"] = self.dataset_details.toJSON()
+        model_card["training_details"] = self.training_details.toJSON()
+        model_card["eval_details"] = self.eval_details.toJSON()
+        model_card["experiment_environment"] = self.experiment_environment.toJSON()
+        return model_card
+
+class SystemEnv:
+
+    @staticmethod
+    def create_model_card(grid_search_id: str, exp_config: dict, gs_config: dict, model = None) -> Dict:
+        """
+        Create Model card.
+        :params:
+                grid_search_id (str): Grid Search ID created for the run.
+                exp_config (dict): Experiment configuration.
+                gs_config (dict): Grid Search configuration.
+        :returns:
+                model_card (ModelCard): Model card object to be converted into json file.
+        """
+        
+        def update_model_details(grid_search_id: str, gs_config: dict, model) -> ModelDetails:
+            """
+            Function to initialize ModelDetails object.
+            :params:
+                    grid_search_id (str): Grid Search ID created for the run.
+                    gs_config (dict): Grid Search configuration.
+            :returns:
+                    obj (ModelDetails): initialized model details object.
+            """
+            try:
+                if model != None:
```

Comment:
wrong way to check for None. 
it has to be: `if model is not None`

Reply:
ok updated it. Thank you.